File: URI_Online_Judge/Problem_1936/program.py
from math import *

#Reads number from input:
n = input()

#Square root of n:
sq = ceil(n**0.5)

#Sum in progress (never bigger than n):
cont = 0

#Answer (number of portions):
ans = 0

#Loops until the sum in progress reaches n:
while cont!=n:
    #Tries to find a number whose factorial is smaller than n-(sum in progress):
	for i in range(int(sq),0,-1):
		if factorial(i) <= n-cont:
			cont+=factorial(i)
			ans+=1
			sq=i
			break
			
print ans

File: URI_Online_Judge/Problem_1936/program.cpp
#include <iostream>
#include <cmath>

using namespace std;


int factorial(int n){
	if(n == 0) return 1;
	if(n == 1) return 1;
	return factorial(n-1) * n;
}

int main(){

	int n;
	cin >> n;

	int curr = ceil(sqrt(n));
	int fact = factorial(curr);

	int acc = 0;
	int answer = 0;

	while (acc != n){
		fact /= curr;
		if (fact < n - acc){
			acc += fact;
			answer += 1;
		}
		curr--;
	}

	cout << answer << endl;

}
